### ISSUE TITLE
Populating a field resets connections in adapter

### ISSUE THREAD

**Issue by KeKs0r** [balderdashy/sails-postgresql#143]
Hi,

I am having an issue with the following code:
```javascript
       Group.find({level: level})
            .populate('children')
            .then(function (model) {
                if (!model) {
                    return reply().code(404);
                }
                reply(model.toJSON());
            })
            .catch(reply);
```
It throws the following error:
```
Debug: internal, implementation, error 
    Error: Uncaught error: Unable to determine primary key for collection `gamegroup` because an error was encountered acquiring the collection definition:
[TypeError: Cannot read property 'collections' of undefined]
    at _getPK (/Users/marc/Sites/hapigamb/node_modules/sails-postgresql/lib/adapter.js:997:13)
    at module.exports.adapter.join.Cursor.$populateBuffers.async.auto.buildChildBuffers (/Users/marc/Sites/hapigamb/node_modules/sails-postgresql/lib/adapter.js:642:28)
    at async.each.callback (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:125:13)
    at Array.forEach (native)
    at _each (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:46:24)
    at Object.async.each (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:124:9)
    at Array.module.exports.adapter.join.Cursor.$populateBuffers.async.auto.buildChildBuffers [as 1] (/Users/marc/Sites/hapigamb/node_modules/sails-postgresql/lib/adapter.js:638:23)
    at async.auto.listener (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:493:46)
    at async.auto.taskComplete (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:444:17)
    at Array.forEach (native)
    at _each (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:46:24)
    at Object.async.auto.taskComplete [as _onImmediate] (/Users/marc/Sites/hapigamb/node_modules/async/lib/async.js:443:13)
    at processImmediate [as _immediateCallback] (timers.js:330:15)
```
It seems to come from the population, but weirdly those 2 commands work:
```javascript
       Group.find({level: level})
            .then(function (model) {
```

```javascript
        Group.findOne(id)
            .populate('children')
            .then(function (model) {
```

Any ideas?

**Comment by devinivy** [comment]
Hm!  It looks like the issue is here: https://github.com/balderdashy/sails-postgresql/blob/master/lib/adapter.js#L982

Do you use custom column or table names?  Mind sharing your group and gamegroup model definitions?

**Comment by sailsbot** [comment]
Thanks for posting, @KeKs0r.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
